src/core/server/internal_types.ts

```diff
@@ -49,6 +49,7 @@ export interface InternalCoreSetup {
  */
 export interface InternalCoreStart {
   capabilities: CapabilitiesStart;
+  elasticsearch: ElasticsearchServiceStart;
```

Comment:
Shouldn't we migrate the platform code to the new API? I see that SO still uses the deprecated one. It can be done as a follow-up.

Reply:
And we should update the migration guide accordingly.

Reply:
> Shouldn't we migrate the platform code to the new API? I see that SO still uses the deprecated one. It can be done as a follow-up.

I've added a task to https://github.com/elastic/kibana/issues/55397